projects/plugins/jetpack/modules/woocommerce-analytics/classes/class-jetpack-woocommerce-analytics-trait.php

```diff
@@ -26,6 +26,132 @@ trait Jetpack_WooCommerce_Analytics_Trait {
 	 */
 	protected $cart_checkout_templates_in_use;
 
+	/**
+	 * The content of the cart page or where the cart page is ultimately derived from if using a template.
+	 *
+	 * @var string
+	 */
+	protected $cart_content_source = '';
+
+	/**
+	 * The content of the checkout page or where the cart page is ultimately derived from if using a template.
+	 *
+	 * @var string
+	 */
+	protected $checkout_content_source = '';
+
+	/**
+	 * Gets the content of the cart/checkout page or where the cart/checkout page is ultimately derived from if using a template.
+	 * This method sets the class properties $checkout_content_source and $cart_content_source.
+	 *
+	 * @return void Does not return, but sets class properties.
+	 */
+	public function find_cart_checkout_content_sources() {
+
+		$cart_checkout_content_cache_transient_name = 'jetpack_woocommerce_analytics_cart_checkout_content_sources';
+
+		$transient_value = get_transient( $cart_checkout_content_cache_transient_name );
+
+		if (
+			false !== $transient_value &&
+			! empty( $transient_value['checkout_content_source'] ) &&
+			! empty( $transient_value['cart_content_source'] )
+		) {
+			$this->cart_content_source     = $transient_value['cart_content_source'];
+			$this->checkout_content_source = $transient_value['checkout_content_source'];
+			return;
+		}
+
```

Comment:
It would help greatly if we have a list here of the steps we go through to get the logic as I'm finding it hard to follow if we already achieved our goal, branched out to something else, or still trying to get the content.